### ISSUE TITLE
Is it possible to use Unet for multi-output prediction?

### ISSUE THREAD

**Issue by Minxiangliu** [Project-MONAI/MONAI#3721]
I have some PET images and have marked the ROI of the cancer range, however I want to use this data to make other predictions, including probabilistic prediction, linear regression, etc., so there is a way to do like what this [blog](https://www.pyimagesearch.com/2018/06/04/keras-multiple-outputs-and-multiple-losses/) mentions `Multiple outputs and multiple losses` architecture?

Thanks in advance!

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Could you please help share some information about this UNet question?

Thanks in advance.

**Comment by Minxiangliu** [comment]
@Nic-Ma  Thank you very much for your quick reply, I want to correct my question a little. How to use the existing model to connect different losses and make different predictions?
EX:
                                   |---Dense().....Dense(sigmoid)
monai.networks.nets  |---Dense().....Dense(sigmoid)
                                   |---Dense().....Dense(linear)

**Comment by ericspod** [comment]
I don't think it's a straight forward question of what to do with the existing UNet architecture to do something like this. The network in the blog is composed of two distinct branches and has two outputs, so the equivalent idea here could be to have distinct UNet instances for the tasks you want, and wrap these in a class that broadcasts the inputs to each. This class is what you'd pass to the training engine, and similarly you could have a wrapper around the loss functions you want to use so that they appear as one. Something like this perhaps:

```py
class NetWrapper(nn.module):
    def __init__(self, nets):
        self.nets=nets
    def forward(self, x):
        return tuple(net(x) for net in nets)

class LossWrapper(nn.module):
    def __init__(self,losses):
        self.losses=losses
    def forward(self, inputs, target):
        return sum(loss(input, grnd) for loss, input, grnd in zip(self.losses, inputs, target))
```
